src/lib/queries/ibc.ts

```diff
@@ -356,6 +290,104 @@ export function setupIbcExtension(base: QueryClient): IbcExtension {
           return response;
         },
       },
+      proof: {
+        channel: {
+          channel: async (portId: string, channelId: string) => {
+            // key: https://github.com/cosmos/cosmos-sdk/blob/ef0a7344af345882729598bc2958a21143930a6b/x/ibc/24-host/keys.go#L117-L120
+            const key = toAscii(
+              `channelEnds/ports/${portId}/channels/${channelId}`
+            );
+            const proven = await base.queryRawProof('ibc', key);
+            const channel = Channel.decode(proven.value);
+            const proof = ProofOps.encode(proven.proof).finish();
+            const proofHeight = Height.fromPartial({
+              revisionHeight: new Long(proven.height),
+            });
+            return {
+              channel: channel,
+              proof: proof,
+              proofHeight: proofHeight,
+            };
+          },
+          packetCommitment: async (
+            portId: string,
+            channelId: string,
+            sequence: number
+          ) => {
+            // key: https://github.com/cosmos/cosmos-sdk/blob/ef0a7344af345882729598bc2958a21143930a6b/x/ibc/24-host/keys.go#L183-L185
+            const key = toAscii(
+              `commitments/ports/${portId}/channels/${channelId}/packets/${sequence}`
+            );
+            const proven = await base.queryRawProof('ibc', key);
+            const commitment = proven.value;
+            const proof = ProofOps.encode(proven.proof).finish();
```

Comment:
Thanks for porting all these over.

Odd that there is the proof field in the grpc (which would seem to fit), but is never filled. So we make raw queries and then construct the grpc result type.

Anyway, great to have these, just grumbling about the sdk